rclcpp/CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  add_compile_options(-Wall -Wextra -Wpedantic)
+  add_compile_options(-Wall -Wextra -Wpedantic -Wnon-virtual-dtor -Woverloaded-virtual)
```

Comment:
@ivanpauno will `-Woverloaded-virtual` result in a warning if said method is both overriden and overloaded?

Reply:
gcc man page explanation is better than mine:

> Warn when a function declaration hides virtual functions from a base class